Teach webpushd/webpushtool about push partition strings

beidson opened the pull request:
#### a1cdbb2c2461d1c597f0826911626b87b41a064a
<pre>
Teach webpushd/webpushtool about push partition strings
<a href="https://bugs.webkit.org/show_bug.cgi?id=249007">https://bugs.webkit.org/show_bug.cgi?id=249007</a>

Reviewed by Ben Nham.

-Use push partition in more places
-Including fleshing out app launch strategies

* Source/JavaScriptCore/Configurations/JavaScriptCore.xcconfig: Fix iOS build.

* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/Platform/IPC/DaemonCoders.cpp:
(WebKit::Daemon::void&gt;::encode):
(WebKit::Daemon::void&gt;::decode):
* Source/WebKit/Platform/spi/ios/FrontBoardServicesSPI.h:
* Source/WebKit/Shared/Cocoa/WebPushMessageCocoa.mm:
(WebKit::WebPushMessage::fromDictionary):
(WebKit::WebPushMessage::toDictionary const):
* Source/WebKit/Shared/PushMessageForTesting.h:
(WebKit::WebPushD::PushMessageForTesting::encode const):
(WebKit::WebPushD::PushMessageForTesting::decode):
* Source/WebKit/Shared/WebPushMessage.h:
* Source/WebKit/Shared/WebPushMessage.serialization.in:
* Source/WebKit/webpushd/LaunchServicesSPI.h:
* Source/WebKit/webpushd/PushClientConnection.h:
(WebPushD::ClientConnection::pushPartitionString const):
* Source/WebKit/webpushd/PushService.mm:
(WebPushD::PushService::didReceivePushMessage):
* Source/WebKit/webpushd/WebPushDaemon.mm:
(WebPushD::Daemon::injectPushMessageForTesting):
(WebPushD::Daemon::notifyClientPushMessageIsAvailable):
(WebPushD::Daemon::getPendingPushMessages):
* Source/WebKit/webpushd/webpushtool/WebPushToolMain.mm:
(pushMessageFromArguments):
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/FrontBoardServices.framework/FrontBoardServices.tbd: Added.

Canonical link: <a href="https://commits.webkit.org/257680@main">https://commits.webkit.org/257680@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d63b36d70781daef355077a6f8db31c97ca3c12d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/99667 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/8844 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/32747 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/109029 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/169260 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/9430 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/86133 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/92128 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/106937 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/105439 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/7107 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/90620 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [💥 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34067 "An unexpected error occured. Recent messages:Cleaned up git repository; Skipping applying patch since patch_id isn't provided; Checked out pull request; Updated gtk dependencies; Pull request contains relevant changes; Skipped layout-tests; layout-tests (exception)") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/88922 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/21979 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/90314 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/2667 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/23495 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/86205 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/2612 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/45880 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/28944 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5284 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/8749 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/42969 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/89078 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/4474 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/19914 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

bnham commented on Source/WebKit/webpushd/webpushtool/WebPushToolMain.mm:
```diff
@@ -77,7 +81,7 @@ static void printUsageAndTerminate(NSString *message)
         return nullptr;
 
-    PushMessageForTesting pushMessage = { appIdentifier, registrationURL, message };
+    PushMessageForTesting pushMessage = { appIdentifier, partitionString, registrationURL, message };
```
Seems like we could just put the partitionString into the call to updateConnectionConfiguration and leave PushMessageForTesting alone?

bnham reviewed (approved):
LGTM after you get the iOS build failures fixed.

bnham commented on Source/WebKit/Configurations/WebKit.xcconfig:
```diff
@@ -171,10 +171,14 @@ WK_INSTALL_COORDINATION_LDFLAGS = $(WK_INSTALL_COORDINATION_LDFLAGS_$(WK_PLATFOR
 WK_INSTALL_COORDINATION_LDFLAGS_iphoneos = -framework InstallCoordination;
 WK_INSTALL_COORDINATION_LDFLAGS_iphonesimulator = $(WK_INSTALL_COORDINATION_LDFLAGS_iphoneos)
 
+WK_FRONTBOARDSERVICES_LDFLAGS = $(WK_FRONTBOARDSERVICES_LDFLAGS_$(WK_PLATFORM_NAME));
```
You might need a tbd in `WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16` for this to build on the bots.

beidson committed:
Teach webpushd/webpushtool about push partition strings
https://bugs.webkit.org/show_bug.cgi?id=249007

Reviewed by Ben Nham.

-Use push partition in more places
-Including fleshing out app launch strategies

* Source/JavaScriptCore/Configurations/JavaScriptCore.xcconfig: Fix iOS build.

* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/Platform/IPC/DaemonCoders.cpp:
(WebKit::Daemon::void>::encode):
(WebKit::Daemon::void>::decode):
* Source/WebKit/Platform/spi/ios/FrontBoardServicesSPI.h:
* Source/WebKit/Shared/Cocoa/WebPushMessageCocoa.mm:
(WebKit::WebPushMessage::fromDictionary):
(WebKit::WebPushMessage::toDictionary const):
* Source/WebKit/Shared/PushMessageForTesting.h:
(WebKit::WebPushD::PushMessageForTesting::encode const):
(WebKit::WebPushD::PushMessageForTesting::decode):
* Source/WebKit/Shared/WebPushMessage.h:
* Source/WebKit/Shared/WebPushMessage.serialization.in:
* Source/WebKit/webpushd/LaunchServicesSPI.h:
* Source/WebKit/webpushd/PushClientConnection.h:
(WebPushD::ClientConnection::pushPartitionString const):
* Source/WebKit/webpushd/PushService.mm:
(WebPushD::PushService::didReceivePushMessage):
* Source/WebKit/webpushd/WebPushDaemon.mm:
(WebPushD::Daemon::injectPushMessageForTesting):
(WebPushD::Daemon::notifyClientPushMessageIsAvailable):
(WebPushD::Daemon::getPendingPushMessages):
* Source/WebKit/webpushd/webpushtool/WebPushToolMain.mm:
(pushMessageFromArguments):
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/FrontBoardServices.framework/FrontBoardServices.tbd: Added.

Canonical link: https://commits.webkit.org/257680@main
